### PR TITLE
Update node pg and node engine.

### PR DESCRIPTION
**Pull request by joeandaverde**
Start upgrade to expose pg pool and perhaps make node pg a peer dependency.

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -1,7 +1,20 @@
...
+   pool_options?: {
+      max?: number
+      min?: number
+      connection_timeout_ms?: number
+      idle_timeout_ms?: number
+      application_name?: string
+   }
+}
+
+export type TinyPgArgumentType = null | string | number | boolean | object | Date | TinyPgArguments
```
Comment (joeandaverde) [inline src/types.ts]: not sure if we want object here?

File: src/tiny.ts [hunk 2]
```diff
@@ -39,34 +39,41 @@ export class TinyPg {
          port: parseInt(params.port, 10),
          database: params.pathname.split('/')[1],
          ssl: params.query.sslmode !== 'disable',
+         connectionTimeoutMillis: pool_options.connection_timeout_ms,
```
Comment (alexmunda) [inline src/tiny.ts]: might as well spell `connectionTimeoutMilliseconds`

File: src/tiny.ts [hunk 3]
```diff
@@ -39,34 +39,41 @@ export class TinyPg {
...
+         idleTimeoutMillis: pool_options.idle_timeout_ms,
```
Comment (alexmunda) [inline src/tiny.ts]: `idleTimeoutMilliseconds`

File: package.json [hunk 4]
```diff
@@ -29,25 +29,28 @@
   ],
   "license": "MIT",
   "engines": {
-    "node": ">= 6.0.0"
+    "node": ">= 8.0.0"
   },
   "dependencies": {
+    "@types/node": "^8.10.0",
```
Comment (alexmunda) [inline package.json]: Should `@types/node` and `@types/pg` be `devDependencies`?

File: src/tiny.ts (hunk 2 above)
Comment (joeandaverde) [reply src/tiny.ts]: The name is designated by the pgpool options type.

File: package.json (hunk 4 above)
Comment (joeandaverde) [reply package.json]: Need to understand how to do this right. I thought I had trouble with installation in a consuming repo if the type wasn’t already included.

**Pull request merged**